src/pq_flash_index.cpp

```diff
@@ -157,6 +157,8 @@ namespace diskann {
         diskann::alloc_aligned((void **) &scratch.aligned_query_float,
```

Comment:
Curious, can the allocations be combined into a single bigger one? It may reduce fragmentation, and also contention (probably not significant though). Not critical for this PR.

Reply:
@hliu18 What contention would they reduce? One advantage of merging this mallocs and making them continuous could be to reduce TLB cache misses (since all this data would be contiguous)

Reply:
No rush. Each alloc/free operation takes lock (critical section). large number of alloc/free operation can cause lock contention.

Reply:
Huisheng, this is only on the setup phase that is run once, not on the search path. Would you still suggest merging the allocs -- that would make the code messy, isn't it?

Reply:
If it only runs once, it's fine.